src/app/components/navbar/navbar.component.html

```diff
@@ -1,11 +1,11 @@
-<nav class="navbar navbar-light justify-content-between m-2">
+<nav class="navbar navbar-light justify-content-between" [ngStyle]="bgStyle" (window:scroll)="addBackground()">
```

Comment:
it's not ok - a lot of events are fired. try to use IntersectionObserver

https://taylor.callsen.me/creating-a-shrinking-logo-animation-with-intersection-observers/
https://blog.logrocket.com/how-to-do-scroll-linked-animations/